### ISSUE TITLE
Add support for HuggingFace models (i.e. HuggingFaceCall)

### ISSUE THREAD

**Issue by brenkao** [Mirascope/mirascope#215]
**Is your feature request related to a problem? Please describe.**
While HuggingFace is already supported through the `OpenAICall` class (use base_url, model, api_key="-"), we should support HuggingFace InferenceClient through a `HuggingFaceCall` class.

**Describe the solution you'd like**
Create a `HugginFaceCall` that extends `BaseCall` that follows same principles as other Calls. This means that call params are properly typed in a Pydantic model, responses extend `BaseCallResponse`, etc.

**Additional context**
See OpenAICall for reference.
https://github.com/Mirascope/mirascope/blob/dev/mirascope/openai/calls.py

**Comment by willbakst** [comment]
The main thing on my mind for this feature is how this will interact with our `prompt_template` parsing. Will users want direct access to the underlying model prompt structure?

e.g. https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-3/

**Comment by willbakst** [comment]
Likely worth looking into how we can propagate the logits of the output as well

**Comment by willbakst** [comment]
Since the Messages API for Text Generation Inference supports the OpenAI SDK, models on HF should be accessible through the `openai` module by setting a custom client.

For now, I'm closing this issue until there is a strong reason to implement more direct support.

See: https://huggingface.co/docs/text-generation-inference/en/basic_tutorials/consuming_tgi#openai-client
Supported Models: https://huggingface.co/docs/text-generation-inference/en/supported_models
